plugins/webp-uploads/helper.php

```diff
@@ -322,3 +328,59 @@ function webp_uploads_should_discard_additional_image_file( array $original, arr
 	}
 	return false;
 }
+
+/**
+ * Checks if a mime type is supported by the server.
+ *
+ * Includes special handling for false positives on AVIF support.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $mime_type The mime type to check.
+ * @return bool Whether the server supports a given mime type.
+ */
+function webp_uploads_mime_type_supported( string $mime_type ): bool {
+	if ( ! wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) ) {
+		return false;
+	}
+
+	// In certain server environments Image editors can report a false positive for AVIF support.
+	if ( 'image/avif' === $mime_type ) {
+		$editor = _wp_image_editor_choose( array( 'mime_type' => 'image/avif' ) );
+		if ( false === $editor ) {
+			return false;
+		}
+		if ( is_a( $editor, WP_Image_Editor_GD::class, true ) ) {
+			return function_exists( 'imageavif' );
+		}
+		if ( is_a( $editor, WP_Image_Editor_Imagick::class, true ) && class_exists( 'Imagick' ) ) {
+			return 0 !== count( Imagick::queryFormats( 'AVIF' ) );
+		}
+	}
+
+	return true;
+}
+
+/**
+ * Get the image output format from the option, falling back to webp if unset.
+ *
+ * @since n.e.x.t
+ *
+ * @return string The image output format. One of 'webp' or 'avif'.
+ */
+function webp_uploads_get_image_output_format(): string {
+	$image_format = get_option( 'perflab_modern_image_format' );
+	return webp_uploads_image_format_is_supported( $image_format ) ? $image_format : 'webp';
+}
+
+/**
+ * Check if an image format type is supported by the plugin.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $image_format The image format to check.
+ * @return bool Whether the image format is supported.
+ */
+function webp_uploads_image_format_is_supported( string $image_format ): bool {
+	return in_array( $image_format, array( 'webp', 'avif' ), true );
+}
```

Comment:
```suggestion
/**
 * Sanitizes the image format.
 *
 * @since n.e.x.t
 *
 * @param string $image_format The image format to check.
 * @return string Supported image format.
 */
function webp_uploads_sanitize_image_format( string $image_format ): string {
	return in_array( $image_format, array( 'webp', 'avif' ), true ) ? $image_format : 'webp';
}
```